File: src/behavioral/visitor/python/classes/medicine.py
from classes.interfaces.product_protocol import VisitableProductProtocol
from classes.interfaces.tax_visitor_protocol import TaxVisitor


class Medicine(VisitableProductProtocol):

    def get_price_with_taxes(self, visitor: TaxVisitor) -> float:
        return visitor.calculate_taxes_for_medicine(self)

File: src/behavioral/visitor/typescript/classes/medicine.ts
import { AbstractVisitableProduct } from './interfaces/AbstractVisitableProduct';
import { VisitorProtocol } from './interfaces/VisitorProtocol';

export class Medicine extends AbstractVisitableProduct {
  getPriceWithTaxes(visitor: VisitorProtocol): number {
    return visitor.taxesForMedicines(this);
  }
}
